packages/front-end/components/Metrics/MetricForm/index.tsx

```diff
@@ -403,6 +421,23 @@ const MetricForm: FC<MetricFormProps> = ({
     name: "conditions",
   });
 
+  const defaultSqlTemplate = selectedDataSource
+    ? getInitialMetricQuery(selectedDataSource, value.type, value.name)[1]
+    : "";
+
+  const resetSqlToDefault = (datasource, type, name) => {
+    if (datasource?.properties?.queryLanguage === "sql" && datasource) {
+      const [userTypes, sql] = getInitialMetricQuery(datasource, type, name);
+      form.setValue("sql", sql);
+      form.setValue("userIdTypes", userTypes);
+      form.setValue("queryFormat", "sql");
```

Comment:
If someone is using the query builder and then they change the metric name, will this switch them back to using SQL instead?  Is that expected?

Reply:
Indeed it does.  This was copied from the old code... but it only did it once setting the default queryFormat.  Seems reasonable that if they ever change it to builder themselves afterwards that we wouldn't want to force them back to sql.  This is a bit tricky to detect because the default value if there is no current.queryFormat nor current.sql is `builder`.  So I can check to see if it has sql set or not and only change the queryFormat then.  But then if someone decides to set the sql back to `""` when changing to query state `builder` then the logic will break.  So I'll add a new state.